pkg/wrangler/context.go

```diff
@@ -246,3 +247,16 @@ func syncOnlyChangedObjects(option controllerContextType) bool {
 	}
 	return false
 }
+
+// GetClusterContext creates a wrangler context to interact with a specific cluster.
+func (w *Context) GetClusterContext(clusterID string) (*Context, error) {
```

Comment:
This looks good Priya, my only feedback is call it `DownStreamClusterContext`, without the get.

Reply:
Thanks for the review Israel :) I updated the function name to `DownStreamClusterWranglerContext`